src/lib/server/tmpl_tokenize.c

```diff
@@ -396,6 +396,34 @@ void tmpl_debug(tmpl_t const *vpt)
  * @{
  */
 
+/** Parse one a single list reference
+ *
+ * @param[out] da_p	attribute representing a list.
+ * @param[in] in	Sbuff to read request references from.
+ * @return
+ *	- > 0 the number of bytes parsed.
+ *      - 0 no list qualifier found.
+ */
+fr_slen_t tmpl_attr_list_from_substr(fr_dict_attr_t const **da_p, fr_sbuff_t *in)
```

Comment:
Why is this needed?  There shouldn't be a special case for list attributes anywhere.

Reply:
Mostly for the unit test utilities

Reply:
There are two places it's used - when compiling `update` sections into `edit` sections and in `_map_afrom_cs()` when parsing `update` sections

Reply:
There is another use in `rlm_exec` but that is easily removable